eng/common/scripts/GeneralReleaseNotesLogic.ps1

```diff
@@ -0,0 +1,128 @@
+. (Join-Path $PSScriptRoot common.ps1)
+
+$PATTERN_REGEX = "^\[\w+(\.(?<SectionName>\w+))?\]:\s#\s\((?<Pattern>.*)\)"
+$CsvMetaData = Get-CSVMetadata
+
+function Get-PackagesInfoFromFile ($releaseNotesLocation) 
+{
+    $releaseNotesContent = Get-Content -Path $releaseNotesLocation
+    $checkLine = $False
+    $presentPkgInfo = @()
+
+    foreach ($line in $releaseNotesContent)
+    {
+        if ($line -eq "<!--")
+        {
+            $checkLine = $True
+            continue
+        }
+        if ($line -eq "-->")
+        {
+            break
+        }
+        if ($checkLine)
+        {
+            $pkgInfo = ($line.Trim()).Split(":")
+            $packageName = $pkgInfo[0]
+            $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+            if ($packageMetaData.Count -gt 0)
+            {
+                $presentPkgInfo += $line.Trim()
+            }
+        }
+    }
+    return $presentPkgInfo
+}
+
+function Filter-ReleaseHighlights ($releaseHighlights)
+{
+    $results = @{}
+
+    foreach ($key in $releaseHighlights.Keys)
+    {
+        $keyInfo = $key.Split(":")
+        $packageName = $keyInfo[0]
+        $packageVersion = $keyInfo[1]
+
+        $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+
+        if ($packageMetaData.ServiceName -eq "template")
+        {
+            continue
+        }
+
+        $existingPackages = GetExistingPackageVersions -PackageName $packageName `
+        -GroupId $packageMetaData.GroupId
+
+        $versionExists = $existingPackages | Where-Object { $_ -eq $packageVersion }
+
+        if ($null -eq $versionExists)
+        {
+            continue
+        }
+
+        $results.Add($key, $releaseHighlights[$key])
+    }
+    return $results
+}
+
+function Write-GeneralReleaseNote ($releaseHighlights, $releaseFilePath)
+{
+    $releaseContent = Get-Content $releaseFilePath
+    $newReleaseContent = @()
+    $writingPaused = $False
+
+    foreach ($line in $releaseContent)
+    {
+        if ($line -match $PATTERN_REGEX)
+        {
+            $sectionName = $matches["SectionName"]
+            $pattern = $matches["Pattern"]
+
+            foreach ($key in $releaseHighlights.Keys)
+            {
+                $pkgInfo = $key.Split(":")
+                $packageName = $pkgInfo[0]
+                $packageVersion = $pkgInfo[1]
+                $packageFriendlyName = ($csvMetaData | Where-Object { $_.Package -eq $PackageName }).DisplayName
```

Comment:
There might end up being 0, 1, or more entries from this filtering. Especially in Java there are often duplicate package names with different groupids.